src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1118,6 +1118,17 @@ void SQLStorage::saveInstalledVersion(const std::string& ecu_serial, const Uptan
       LOG_ERROR << "Failed to save installed versions: " << db.errmsg();
       return;
     }
+  } else if (update_mode == InstalledVersionUpdateMode::kBadTarget) {
```

Comment:
At first glance, it looks like this `if` is better suited in the `if else` below before `if (!!old_id)`.
```
if (update_mode == InstalledVersionUpdateMode::kBadTarget) {
...
} else if (!!old_id) {
...
```
What do you think?

Reply:
Sorry, missed this comment. It makes sense, to avoid doing 2 updates. I'm testing the change and will update the PR accordingly.